cmd/juju/commands/upgradecontroller.go

```diff
@@ -268,7 +268,13 @@ func (c *upgradeControllerCommand) upgradeIAASController(ctx *cmd.Context) error
 	}}
 	jcmd.SetClientStore(c.ClientStore())
 	wrapped := modelcmd.Wrap(jcmd)
-	args := append(c.rawArgs, "-m", bootstrap.ControllerModelName)
+	controllerName, err := c.ControllerName()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	fullControllerModelName := modelcmd.JoinModelName(controllerName,
+		bootstrap.ControllerModelName)
```

Comment:
Um... I don't think this will work if someone actually had their controller model named differently.
Now, I don't think anyone ever does change the name of the controller model, but they can.
Is there a way to get the information out of the local store?

Reply:
Probably, but it looks like ControllerModelName is relied upon heavily in other commands for finding the controller model uuid. So perhaps this is for another chunk of work.

Reply:
I agree that if anyone has a controller model named differently, this code could be problematic but I also agree that there may be other issues and commands that will fail in that scenario. Could you please file a bug?